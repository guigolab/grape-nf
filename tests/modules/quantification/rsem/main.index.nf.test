nextflow_process {

    name "Test Process index"
    script "modules/quantification/rsem/main.nf"
    process "index"

    tag "module"

    test("Should create index for RSEM quantification (uncompressed references)") {

        when {
            process {
                """
                input[0] = file("${baseDir}/data/genome.fa")
                input[1] = file("${baseDir}/data/annotation.gtf")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Should create index for RSEM quantification (compressed references)") {

        when {
            process {
                """
                input[0] = file("${baseDir}/data/genome.fa.gz")
                input[1] = file("${baseDir}/data/annotation.gtf.gz")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
