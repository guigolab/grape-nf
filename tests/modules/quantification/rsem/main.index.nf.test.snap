{
    "Should create index for RSEM quantification (compressed references)": {
        "content": [
            {
                "0": [
                    [
                        "RSEMref.chrlist:md5,5c6cc18be0eab93c65b0cf7d65ccdf40",
                        "RSEMref.grp:md5,0f456a742fad8dbc1e1f042ba6d2eb80",
                        "RSEMref.idx.fa:md5,01a65f27053fc82b169cb3856c882314",
                        "RSEMref.n2g.idx.fa:md5,01a65f27053fc82b169cb3856c882314",
                        "RSEMref.seq:md5,a774142fe7d5764339a5302cb581791e",
                        "RSEMref.ti:md5,32bd9a68c9ab39fbbafc6a9bb59b9778",
                        "RSEMref.transcripts.fa:md5,36e748ee42105667941d667db18c651d"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-03T16:01:02.486991"
    },
    "Should create index for RSEM quantification (uncompressed references)": {
        "content": [
            {
                "0": [
                    [
                        "RSEMref.chrlist:md5,5c6cc18be0eab93c65b0cf7d65ccdf40",
                        "RSEMref.grp:md5,0f456a742fad8dbc1e1f042ba6d2eb80",
                        "RSEMref.idx.fa:md5,01a65f27053fc82b169cb3856c882314",
                        "RSEMref.n2g.idx.fa:md5,01a65f27053fc82b169cb3856c882314",
                        "RSEMref.seq:md5,a774142fe7d5764339a5302cb581791e",
                        "RSEMref.ti:md5,32bd9a68c9ab39fbbafc6a9bb59b9778",
                        "RSEMref.transcripts.fa:md5,36e748ee42105667941d667db18c651d"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-03T16:00:53.032961"
    }
}